awspub/notification.py

```diff
@@ -0,0 +1,94 @@
+"""
+Methods used to handle notifications for AWS using SNS
+"""
+
+import json
+import logging
+from typing import Any, Dict
+
+import boto3
+from mypy_boto3_sns.client import SNSClient
+
+from awspub.exceptions import AWSNotificationException
+
+logger = logging.getLogger(__name__)
+
+
+class SNSNotification(object):
+    """
+    A data object that contains validation logic and
+    structuring rules for SNS notification JSON
+    """
+
+    def __init__(self, conf: Dict[str, Any], image_id: str, region_name: str):
+        """
+        Construct a message and verify that it is valid
+        """
+        self._conf = conf
+        self._image_id = image_id
+        self._region_name = region_name
+
+    def _get_topic(self, client: SNSClient) -> str:
+        """
+        Get topic ARN
+
+        :param client: AWS boto3 client with SNS service
+        :type client: SNSClient
+        :return: return topic ARN or empty string if topic does not exist
+        :rtype: str
+        """
+        resp = client.list_topics()
+        for topic in resp["Topics"]:
+            # Matching topic name at the end of ARN
+            if self.topic_name in topic["TopicArn"]:
```

Comment:
if you name your topic `sns`, then every topic would match (example name `arn:aws:sns:eu-central-1:887450378614:testtopic`). you need to be more specific . I think splitting by the last `:`

Reply:
changed get_topic_arn